### PR TITLE
RSDK-4033, RSDK-4043 - Add simple module example, streamline arm example + other updates

### PR DESCRIPTION
**Pull request by cheukt**
* Made the current example the complex example, added a single file sensor example. Single file does not use an __init__.py.
* The arm example in jupyter notebook is now single file and does not depend on a kinematics.json file.
* Made the env consistent between examples and jupyter notebook (requirements.txt) and removes the use of poetry (extra dependency)
* For the jupyter example, reworded a bunch of stuff but mostly to make it consistent with the simple example and also point to the complex example when necessary.

* tested by downloading a fresh copy of the repo on a pi and just adding it to the config.

still a lot of improvements that can be made, but wanted to get something out for now
cc @Fahmina @michaellee1019 @dannenberg

### CODE REVIEW

**Review by dannenberg — APPROVED** [review]
huge step up. thrilled to see this!

File: examples/complex_module/README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+# VIAM Complex Module Example
 This example goes through how to create custom modular resources using Viam's python SDK, and how to connect it to a Robot.
 
 This is a limited document. For a more in-depth understanding of modules, see the [documentation](https://docs.viam.com/program/extend/modular-resources/).
 
 ## Purpose
 Modular resources allows you to define custom components and services, and add them to your robot. Viam ships with many component types, but you're not limited to only using those types -- you can create your own using modules.
 
-For more information, see the [documentation](https://docs.viam.com/program/extend/modular-resources/).
+For more information, see the [documentation](https://docs.viam.com/program/extend/modular-resources/). For a simpler example, take a look at the [simple module example](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/simple_module), which only contains one custom resource model in one file.
```
[the remark this answers is not in the document]
Comment (cheukt) [reply examples/complex_module/README.md:9]: the readme here is part of the complex_module example folder, so I feel like unless we want to restructure the examples even more, it's the user's choice as to which example they start with (the line here is just to remind them there's something simpler)

**Review by purplenicole730 — APPROVED** [review]
This is amazing! LGTM

File: examples/complex_module/run.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+cd `dirname $0`
+
+# Create a virtual environment to run our code
+VENV="venv"
```
[the remark this answers is not in the document]
Comment (cheukt) [reply examples/complex_module/run.sh]: ya, updated

File: examples/complex_module/run.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+cd `dirname $0`
+
+# Create a virtual environment to run our code
+VENV_NAME="venv"
```
Comment (njooma) [inline examples/complex_module/run.sh:5]: Does this not assume you have `venv` installed?  and a virtual env named `venv`?

Comment (cheukt) [reply examples/complex_module/run.sh:5]: venv is baked into python 3.4 and up, so should be good for the versions of python we support.
I've used `venv` for myself for the longest time haha, is there a different name you like?

Comment (njooma) [reply examples/complex_module/run.sh:5]: AH sorry this is creating a virtual env. I guess then it assumes that there isn't already a virtual env named `venv`? What would happen if a user already have a `venv`?

Comment (njooma) [reply examples/complex_module/run.sh:5]: NVM, it creates it in this directory, so there should be no collisions.

**Review by njooma — APPROVED** [review]

**Comment by michaellee1019** [comment]
LGTM! This simple example will help a lot. Thanks for reorganizing it.

File: examples/simple_module/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# VIAM Simple Module Example
+This example goes through how to create custom modular resources using Viam's python SDK, and how to connect it to a Robot.
+
+This is a limited document. For a more in-depth understanding of modules, see the [documentation](https://docs.viam.com/program/extend/modular-resources/).
+
+## Purpose
+Modular resources allows you to define custom components and services, and add them to your robot. Viam ships with many component types, but you're not limited to only using those types -- you can create your own using modules.
```
Comment (npentrel) [inline examples/simple_module/README.md:7]:
```suggestion
Modular resources allow you to define custom components and services, and add them to your robot. Viam ships with many component types, but you're not limited to only using those types -- you can create your own using modules.
```

File: examples/simple_module/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+For more information, see the [documentation](https://docs.viam.com/program/extend/modular-resources/). For a more complex example, take a look at the [complex module example](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/complex_module), which only contains multiple new APIs and custom resource models.
```
Comment (npentrel) [inline examples/simple_module/README.md:9]:
```suggestion
For more information, see the [documentation](https://docs.viam.com/program/extend/modular-resources/). For a more complex example, take a look at the [complex module example](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/complex_module), which contains multiple new APIs and custom resource models.
```
It's more complex so 'only' seems off

File: examples/simple_module/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+## Project structure
+The definition of the new resources are in the `main` file of the `src` directory.
+
+The `main.py` file contains the definition of a new sensor model and then registers it. It then creates and starts a module. This file is called by the `run.sh` script, which is the entrypoint for this module. Read further to learn how to connect this module to your robot.
+
+Outside the `src` directory, there is the `client.py` file. This can be used to test the module once it's connected to the robot. You will have to update the credentials and robot address in that file.
```
Comment (npentrel) [inline examples/simple_module/README.md:16]:
```suggestion
Outside the `src` directory, there is a `client.py` file. You can use this file to test the module once you have connected to your robot and configured the module. You will have to update the credentials and robot address in that file.
```

File: examples/simple_module/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+## How to use
```
Comment (npentrel) [inline examples/simple_module/README.md:18]:
```suggestion
## Configuring and using the module
```

File: examples/simple_module/README.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+These steps assume that you have a robot available at [app.viam.com](app.viam.com).
+
+The `run.sh` script is the entrypoint for this module. To connect this module with your robot, you must add this module's entrypoint to the robot's config. For example, this could be `/home/viam-python-sdk/examples/simple_module/run.sh`. See the [documentation](https://docs.viam.com/program/extend/modular-resources/#use-a-modular-resource-with-your-robot) for more details.
```
[the remark this answers is not in the document]
Comment (npentrel) [reply examples/simple_module/README.md:21]: Is that correct?

**Review by npentrel — APPROVED** [review]
Some small suggestions to improve wording but looks good! Sorry for the delay!

File: examples/simple_module/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (cheukt) [reply examples/simple_module/README.md]: changed `your file` to the `entrypoint file`, but ya, that's correct

**Pull request merged**